packages/core/src/external/carequality/ihe-gateway-v2/report/report.ts

```diff
@@ -0,0 +1,321 @@
+import { Pool } from "pg";
+import { z } from "zod";
+import dayjs from "dayjs";
+import {
+  OutboundPatientDiscoveryResp,
+  XCPDGateway,
+  OutboundDocumentQueryResp,
+  XCAGateway,
+  OutboundDocumentRetrievalResp,
+  outboundPatientDiscoveryRespSchema,
+  outboundDocumentQueryRespSchema,
+  outboundDocumentRetrievalRespSchema,
+} from "@metriport/ihe-gateway-sdk";
+
+const patientDiscoveryReportRequests = 10;
+const documentQueryReportRequests = 100;
+const documentRetrievalReportRequests = 100;
+
+type Report = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+  errorOids: {
+    oid: string;
+    error: string;
+  }[];
+};
+
+type SimplifiedReport = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+};
+
+type ReportSummary = {
+  simplifiedReports: SimplifiedReport[];
+  averageSuccessPercentage: number;
+  averageFailurePercentage: number;
+  errorDetailsMap: { [key: string]: { count: number; details: string } };
+};
+
+export const patientDiscoveryResultRowSchema = z.object({
+  data: outboundPatientDiscoveryRespSchema,
+});
+
+export const documentQueryResultRowSchema = z.object({
+  data: outboundDocumentQueryRespSchema,
+});
+
+export const documentRetrievalResultRowSchema = z.object({
+  data: outboundDocumentRetrievalRespSchema,
+});
+
+enum Tables {
+  PatientDiscoveryResult = "patient_discovery_result",
+  DocumentQueryResult = "document_query_result",
+  DocumentRetrievalResult = "document_retrieval_result",
```

Comment:
nit: props/enum items should be camelCase; PascalCase is for Classes/Types only.